sci-mathematics/octave/octave-9.1.0.ebuild

```diff
@@ -0,0 +1,273 @@
+# Copyright 1999-2024 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=8
+
+inherit autotools flag-o-matic fortran-2 java-pkg-opt-2 pax-utils qmake-utils toolchain-funcs xdg
+
+DESCRIPTION="High-level interactive language for numerical computations"
+HOMEPAGE="https://www.gnu.org/software/octave/"
+SRC_URI="mirror://gnu/${PN}/${P}.tar.xz"
+
+LICENSE="GPL-3"
+SLOT="0/${PV}"
+KEYWORDS="~amd64 ~arm ~arm64 ~hppa ~ppc ~ppc64 ~riscv ~x86 ~amd64-linux ~x86-linux"
+
+IUSE="curl doc fftw fltk +glpk gnuplot gui hdf5 imagemagick java json klu opengl portaudio postscript +qhull +qrupdate +qt5 qt6 readline sndfile +sparse spqr ssl sundials X zlib"
+
+# Although it is listed in INSTALL.OCTAVE as a build tool, Octave runs
+# "makeinfo" from sys-apps/texinfo at runtime to convert its texinfo
+# documentation to text (see scripts/help/help.m).
+#
+# (un)zip isn't mentioned, but there's a test that uses it (bug #775254).
+#
+# The use of USE=imagemagick to pull in media-gfx/graphicsmagick is not
+# ideal, but both "graphicsmagick" and "imagemagick" are global USE
+# flags whose existing descriptions conflict with the obvious way we
+# would want to use them in octave. In any case, upstream doesn't really
+# support imagemagick, only graphicsmagick (bug 864785).
```

Comment:
Is this still the case? I find the graphicsmagick/imagemagick situation here a bit confusing.

Reply:
AFAIK. GraphicsMagick has stable API and is careful about ABI, while ImageMagick does whatever whenever. GM is much nicer to have as a dependency.

Reply:
https://bugs.gentoo.org/864785#c13 as documented by you: I would keep current configuration (i.e. only graphicsmagick). It's really confusing the USE=imagemagick but I guess that is out of the scope of the current PR

Reply:
IIRC I went with `USE=imagemagick` because the description of `USE=graphicsmagick` is,

> graphicsmagick - Build and link against GraphicsMagick instead of ImageMagick (requires USE=imagemagick if optional)

and I thought it would be pretty silly to have both flags and then the necessary `REQUIRED_USE="imagemagick? ( graphicsmagick )"` to force the right implementation.